pkg/controller/reconciler/recipe_webhook_test.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"strings"
+	"testing"
+	"time"
+
+	radappiov1alpha3 "github.com/radius-project/radius/pkg/controller/api/radapp.io/v1alpha3"
+	portableresources "github.com/radius-project/radius/pkg/rp/portableresources"
+	"github.com/radius-project/radius/test/testcontext"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+const (
+	defaultNamespace    = "default"
+	validResourceType   = "Applications.Core/extenders"
+	invalidResourceType = "Applications.Core/invalidType"
+)
+
+// Test_ValidateRecipe_Type tests a recipe with valid and invalid types.
+func Test_ValidateRecipe_Type(t *testing.T) {
+	ctx := testcontext.New(t)
+	radius, client := setupWebhookTest(t)
+
+	// Environment is created.
+	createEnvironment(radius, "default")
+
+	t.Run("test recipe for invalid type", func(t *testing.T) {
+		recipeName := "test-recipe-invalidtype"
+		namespace := types.NamespacedName{Namespace: defaultNamespace, Name: recipeName}
+		recipe := makeRecipe(namespace, invalidResourceType)
+		expectedError := fmt.Sprintf("Recipe.radapp.io \"%s\" not found", recipeName)
+
+		err := client.Create(ctx, &corev1.Namespace{ObjectMeta: ctrl.ObjectMeta{Name: namespace.Name}})
+		require.NoError(t, err)
+
+		// We do not check for error here since error may be captured after this initial call is completed.
+		_ = client.Create(ctx, recipe)
+
+		current := &radappiov1alpha3.Recipe{}
+		require.Eventually(t, func() bool {
+			if err := client.Get(ctx, namespace, current); err != nil {
+				require.EqualError(t, err, expectedError)
```

Comment:
Can you use `apierrors.IsNotFound` instead? Example: https://github.com/radius-project/radius/blob/9bcf33fa358383a1c4c54b41753bd79265f41aca/pkg/daprrp/datamodel/daprdetector.go#L42

Error strings can change all of the time, because the intended audience is users. Checking for types is much more resiliant.

Reply:
Note: you'll want to add an import alias like:

```
import apierrors "k8s.io/apimachinery/pkg/api/errors"
```

Reply:
nice! thank u! updated.